tests/nodes/test_invoker.py

```diff
@@ -130,3 +171,100 @@ def has_executed_all(g: GraphExecutionState):
     assert g.is_complete()
 
     assert all((i in g.errors for i in g.source_prepared_mapping["1"]))
+
+
+def test_can_create_batch(mock_invoker: Invoker, simple_graph, simple_batch):
+    batch_process_res = mock_invoker.services.batch_manager.create_batch_process(
+        batch=simple_batch,
+        graph=simple_graph,
+    )
+    assert batch_process_res.batch_id
+    assert len(batch_process_res.session_ids) == 25
+    # TODO: without the mock events service emitting the `graph_execution_state` events,
```

Comment:
You should fix this before merge =).

Also, I'd suggest you add a test that involves subgraphs (Graph nodes), to ensure full node paths work correctly and don't break over time.

Reply:
I'm in a bit over my head on getting the mock events service to emit events that the batch manager service can accept. I tried using the actual FastAPIEventsService, but that needs an event loop to work. Am I missing something simple here?